src/v/kafka/requests/fetch_request.h

```diff
@@ -45,10 +45,10 @@ struct fetch_request final {
 
     struct partition {
         model::partition_id id;
-        int32_t current_leader_epoch; // >= v9
+        int32_t current_leader_epoch{-1}; // >= v9
         model::offset fetch_offset;
         // inter-broker data
-        model::offset log_start_offset; // >= v5
+        model::offset log_start_offset{-1}; // >= v5
         int32_t partition_max_bytes;
```

Comment:
do we want to default initialize the other structs? i think it might be worth auditing the reest.